app/uk/gov/hmrc/alcoholdutyreturns/service/ReturnsService.scala

```diff
@@ -70,7 +73,54 @@ class ReturnsService @Inject() (
                                       returnToSubmit,
                                       returnId.appaId
                                     )
-      _                          <- EitherT.right[ErrorResponse](cacheRepository.clearUserAnswersById(returnId))
+      _                          <- EitherT.right[ErrorResponse](
+                                      cacheRepository
+                                        .get(returnId)
+                                        .map {
+                                          case Some(ua) =>
+                                            auditReturnSubmitted(
+                                              Some(ua),
+                                              returnToSubmit,
+                                              AdrReturnCreatedDetails.fromReturnCreatedDetails(returnCreatedDetails),
+                                              returnId
+                                            )
+                                          case None     =>
+                                            logger.warn("User answers couldn't be retrieved while auditing return submission")
+                                            auditReturnSubmitted(
+                                              None,
+                                              returnToSubmit,
+                                              AdrReturnCreatedDetails.fromReturnCreatedDetails(returnCreatedDetails),
+                                              returnId
+                                            )
+                                        }
+                                        .recoverWith { case e: Throwable =>
+                                          Future.successful(logger.warn("Encountered error while auditing return submission", e))
+                                        }
+                                        .andThen { case _ =>
+                                          cacheRepository.clearUserAnswersById(returnId)
```

Comment:
this should be just the original
_ <- EitherT.right[ErrorResponse](cacheRepository.clearUserAnswersById(returnId))

The point of the for comprehension is to avoid this to make it more obvious what's going on (and only run it if it's been successful so far which in the case of the recovery will be the case)

Would keep andThen etc for things that relate to the current step where functions are being joined together.

in this case clearing the cache is a very separate step.